src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -248,8 +245,7 @@ public void Abort()
                         ResponseBody = _responseBody;
                         DuplexStream = new FrameDuplexStream(RequestBody, ResponseBody);
 
-                        _requestAbortCts = CancellationTokenSource.CreateLinkedTokenSource(_disconnectCts.Token);
-                        RequestAborted = _requestAbortCts.Token;
+                        RequestAborted = _disconnectOrAbortedCts.Token;
```

Comment:
@benaadams It avoids extra synchronization logic that might be required to avoid canceling an already disposed. `_requestAbortCts` to cancel when called from places like `Connection.Abort()`.

Turns out that there is still an issue with canceling a disposed `_disconnectCts`, but at least that isn't being disposed after every request.